morpheus/models/dfencoder/ae_feature_selector.py

```diff
@@ -0,0 +1,538 @@
+            to remain in consideration. Default 0.
+        null_threshold (float): Maximum proportion of null values a column can contain. Default 0.3.
+        cardinality_threshold_high (float): Maximum proportion
+            of cardinality to length of data allowable. Default 0.99.
+        cardinality_threshold_low_n (int): Minimum cardinalty for a feature to be considered numerical
+            during type infernce. Default 10.
+        categorical_features (list[str]): List of features in the data to be considered categorical. Default [].
+        numeric_features (list[str]): List of features in the data to be considered numeric. Default [].
+        ablation_epochs (int): Number of epochs to train the autoencoder.
+        device (str): Device to run the model on, defaults to 'cuda' if available.
+
+    Methods:
+        train_autoencoder(data_loader): Trains the autoencoder using the provided data loader.
+        calculate_loss(model, data_loader): Calculates reconstruction loss using the trained model.
+        preprocess_data(): Preprocesses the DataFrame to handle numerical and categorical data appropriately.
+        remove_low_variance(processed_data): Removes features with low variance.
+        remove_high_correlation(data): Removes highly correlated features based on a threshold.
+        feature_importance_evaluation(processed_data): Evaluates feature importance using the autoencoder.
+        select_features(k_min, k_max): Selects features based on importance, adhering to min/max constraints.
+    """
+
+    def __init__(
+            self,
+            input_json,
+            id_column=None,
+            timestamp_column=None,
+            encoding_dim=None,
+            batch_size=256,
+            variance_threshold=0,
+            null_threshold=0.3,
+            cardinality_threshold_high=0.999,
+            cardinality_threshold_low_n=10,
+            categorical_features=[],
+            numeric_features=[],
+            ablation_epochs=20,
+            device="cuda" if torch.cuda.is_available() else "cpu",
+    ):
+        self.df = pd.json_normalize(input_json)
+        self.df_orig = pd.json_normalize(input_json)
+        self.encoding_dim = encoding_dim or self.df.shape[1] // 2
+        self.batch_size = batch_size
+        self.device = device
+        self.preprocessor = None  # To store the preprocessor for transforming data
+        self.variance_threshold = variance_threshold
+        self.null_threshold = null_threshold
+        self.cardinality_threshold_high = cardinality_threshold_high
+        self.cardinality_threshold_low = cardinality_threshold_low_n
+        self.categorical_features = categorical_features
+        self.numeric_features = numeric_features
+        self.ablation_epochs = ablation_epochs
+        self.id_col = id_column
+        self.ts_col = timestamp_column
+        self.final_features = self.df.columns.tolist()  # Initialize with all features
```

Comment:
Are all of these properties which can be changed by the user of the class? If any of these should not be changed, convert them to private variables with a leading `_`